service/collector/nat.go

```diff
@@ -0,0 +1,179 @@
+package collector
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/sync/errgroup"
+
+	clientaws "github.com/giantswarm/aws-operator/client/aws"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+)
+
+const (
+	labelVPC = "vpc"
+	labelAZ  = "availability_zone"
+)
+
+const (
+	subsystemNAT = "nat"
+)
+
+var (
+	natDesc *prometheus.Desc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, subsystemNAT, "info"),
+		"NAT limits information.",
+		[]string{
+			labelAccountID,
+			labelVPC,
+			labelAZ,
+		},
+		nil,
+	)
+)
+
+type NATConfig struct {
+	Helper *helper
+	Logger micrologger.Logger
+
+	InstallationName string
+}
+
+type NAT struct {
+	helper *helper
+	logger micrologger.Logger
+
+	installationName string
+}
+
+func NewNAT(config NATConfig) (*NAT, error) {
+	if config.Helper == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Helper must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	if config.InstallationName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.InstallationName must not be empty", config)
+	}
+
+	v := &NAT{
+		helper: config.Helper,
+		logger: config.Logger,
+
+		installationName: config.InstallationName,
+	}
+
+	return v, nil
+}
+
+func (v *NAT) Collect(ch chan<- prometheus.Metric) error {
+	awsClientsList, err := v.helper.GetAWSClients()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	var g errgroup.Group
+
+	for _, item := range awsClientsList {
+		awsClients := item
+
+		g.Go(func() error {
+			err := v.collectForAccount(ch, awsClients)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			return nil
+		})
+	}
+
+	err = g.Wait()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (v *NAT) Describe(ch chan<- *prometheus.Desc) error {
+	ch <- natDesc
+	return nil
+}
+
+func (v *NAT) collectForAccount(ch chan<- prometheus.Metric, awsClients clientaws.Clients) error {
+	accountID, err := v.helper.AWSAccountID(awsClients)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	iv := &ec2.DescribeVpcsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name: aws.String("tag:giantswarm.io/organization"),
```

Comment:
Let's use [the constant of the key package](https://github.com/giantswarm/aws-operator/blob/9fed5ecf774592fe2992f538f122e0c4f976b4af/service/controller/key/cluster.go#L40) here.